src/app/shell/details/details.component.ts

```diff
@@ -71,14 +74,14 @@ export class DetailsComponent implements OnInit, OnDestroy {
   }
 
   private setDataSubscribtion(): void {
-    combineLatest([this.isMobileScreen$, this.role$, this.workshop$, this.provider$])
+    combineLatest([this.isMobileScreen$, this.role$, this.workshop$, this.provider$, this.currentProvider$])
       .pipe(takeUntil(this.destroy$))
-      .subscribe(([isMobileScreen, role, workshop, provider]) => {
+      .subscribe(([isMobileScreen, role, workshop, provider, currentProvider]) => {
         this.isMobileScreen = isMobileScreen;
         this.role = role;
         this.workshop = workshop;
         this.provider = provider;
-
+        this.currentProvider = currentProvider;
```

Comment:
As we aren't doing particular logic with the state properties, could we leave the subscription only for required ones and pass other using `async` pipe in template?